backend/canisters/airdrop_bot/impl/src/model/airdrops.rs

```diff
@@ -123,49 +121,61 @@ impl Airdrops {
             }
         }
 
-        if total_tickets == 0 || total_shares == 0 {
+        if ticket_holders.is_empty() || total_shares == 0 {
             return None;
         }
 
         let fund = config.main_chat_fund;
         let prizes = config.lottery_prizes.len();
-        let outcome = AirdropOutcome {
-            participants: user_shares
-                .into_iter()
-                .map(|(u, chit, shares)| {
-                    (
-                        u,
-                        Participant {
-                            chit,
-                            shares,
-                            prize: if shares > 0 {
-                                Some(Prize {
-                                    chat_won: (fund * shares as u128) / total_shares as u128,
-                                    block_index: None,
-                                })
-                            } else {
-                                None
-                            },
-                        },
-                    )
-                })
-                .collect(),
-            lottery_winners: (0..prizes)
-                .map(|i| {
-                    let winning_ticket = (rng.next_u32() % total_tickets) as usize;
-                    let winner = ticket_holders[winning_ticket];
-                    (
-                        winner,
-                        Prize {
-                            chat_won: config.lottery_prizes[i],
-                            block_index: None,
+
+        let participants = user_shares
+            .into_iter()
+            .map(|(u, chit, shares)| {
+                (
+                    u,
+                    Participant {
+                        chit,
+                        shares,
+                        prize: if shares > 0 {
+                            Some(Prize {
+                                chat_won: (fund * shares as u128) / total_shares as u128,
+                                block_index: None,
+                            })
+                        } else {
+                            None
                         },
-                    )
-                })
-                .collect(),
-        };
+                    },
+                )
+            })
+            .collect();
+
+        let mut lottery_winners = Vec::new();
 
-        let airdrop = Airdrop { config, outcome };
+        for i in 0..prizes {
+            let winning_ticket = (rng.next_u32() % ticket_holders.len() as u32) as usize;
+
+            let winner = ticket_holders.remove(winning_ticket);
+
+            lottery_winners.push((
+                winner,
+                Prize {
+                    chat_won: config.lottery_prizes[i],
+                    block_index: None,
+                },
+            ));
+
+            if ticket_holders.is_empty() {
```

Comment:
This needs to be at the start of the loop because ticket_holders could start empty